lib/src/over_react_test/jacket.dart

```diff
@@ -89,28 +91,77 @@ class TestJacket<T extends react.Component> {
     _render(reactElement);
   }
 
-  /// Returns the mounted React component instance.
+  /// Returns the mounted React composite component instance.
+  ///
+  /// > If you are rendering a function component using [mount], calling [getInstance] will throw a `StateError`.
+  /// >
+  /// > * If you are trying to access an instance rendered by the function component:
+  /// >   * Use `forwardRef` to pass refs through the tree.
+  /// > * If you are trying to access / query for a DOM node rendered by the function component, try using:
+  /// >
+  /// >   ```
+  /// >   queryByTestId(jacket.mountNode, yourTestId)
+  /// >   ```
   ReactComponent getInstance() {
-    if (!react_test_utils.isCompositeComponent(_renderedInstance)) {
-      throw UnsupportedError('Not a composite component');
+    if (!_isCompositeComponent) {
+      throw StateError(
+          'getInstance() is only supported when the rendered object is a composite (class based) component. '
+          'If you are rendering a function component, and are trying to:\n\n'
+          '1. Access an instance rendered by the function component: Use `forwardRef` to pass refs through the tree.\n'
+          '2. Access / query for a DOM node rendered by the function component: try using\n'
+          '    queryByTestId(jacket.mountNode, yourTestId)');
     }
 
     return _renderedInstance as ReactComponent;
   }
 
-  /// Returns the props associated with the mounted React component instance.
+  /// Returns the props associated with the mounted React composite component instance.
+  ///
+  /// > If you are rendering a function component using [mount], calling [getProps] will throw a `StateError`.
+  /// >
+  /// > See [getInstance] for more information about this limitation.
   Map getProps() {
     return over_react.getProps(getInstance());
   }
 
-  /// Returns the DOM node associated with the mounted React component instance.
+  /// Returns the DOM node associated with the mounted React composite component instance.
+  ///
+  /// > If you are rendering a function component using [mount], calling [getNode] will throw a `StateError`.
+  /// >
+  /// > Try using one of the following instead:
+  /// >
+  /// > ```
+  /// > jacket.mountNode.querySelector(someSelectorThatTargetsTheRootNode);
+  /// > ```
+  /// >
+  /// > ```
+  /// > queryByTestId(jacket.mountNode, yourRootNodeTestId);
+  /// > ```
   Element getNode() {
+    if (!_isCompositeComponent) {
+      throw StateError(
+          'getNode() is only supported when the rendered object is a composite (class based) component. '
```

Comment:
This should not throw when dealing with DOM components (e.g., `mount(Dom.div()())`).

Reply:
Can we add tests for these errors, asserting that these errors are thrown when we expect them to and not thrown when we don't expect them to, for the different component types? (Composite, function, DOM, fragment / primitive value)